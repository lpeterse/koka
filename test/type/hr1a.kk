fun xauto(f:forall<a> a -> a) {
  f(f)
}

fun auto(f:forall<a> a -> a) : (forall<a> a -> a)
{
  f(f)
}

val ids   = single(id:forall<a> a -> a)  //rigid annotation
val test3  = ids.map(auto)
