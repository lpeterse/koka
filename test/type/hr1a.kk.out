auto      : (f : forall<a> (a) -> a) -> (forall<a> (a) -> a)
ids       : list<forall<a> (a) -> a>
test3     : list<forall<a> (a) -> a>
xauto     : forall<a> (f : forall<b> (b) -> b) -> ((a) -> a)
